packages/dds/tree/src/simple-tree/jsonSchema.ts

```diff
@@ -0,0 +1,208 @@
+import type { SimpleNodeSchemaKind } from "./simpleSchema.js";
+
+/**
+ * The fully-qualified {@link TreeNodeSchemaCore.identifier}.
+ * @example Schema `com.myapp.foo` would be referenced via `#/$defs/com.myapp.foo`.
+ * @alpha
+ */
+export type JsonSchemaId = string;
+
+/**
+ * Reference string pointing to a definition in the schema.
+ * Should be the fully-qualified {@link TreeNodeSchemaCore.identifier}.
+ * @remarks Of the form `#/$defs/<schema-identifier>`, where the `schema-identifier` is the fully-qualified {@link TreeNodeSchemaCore.identifier}.
+ * @example Schema `com.myapp.foo` would be referenced via `#/$defs/com.myapp.foo`.
+ * @alpha
+ */
+export type JsonRefPath = `#/$defs/${JsonSchemaId}`;
+
+/**
+ * JSON entity type.
+ * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-instance-data-model}.
+ * @alpha
+ */
+export type JsonSchemaType = "object" | "array" | JsonLeafSchemaType;
+
+/**
+ * JSON primitive types.
+ * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-instance-data-model}.
+ * @alpha
+ */
+export type JsonLeafSchemaType = "string" | "number" | "boolean" | "null";
+
+/**
+ * Base interface for node schemas represented in {@link https://json-schema.org/draft/2020-12/json-schema-core | JSON Schema} format.
+ * @alpha
+ */
+export interface NodeJsonSchemaBase<
+	TNodeKind extends SimpleNodeSchemaKind,
+	TJsonSchemaType extends JsonSchemaType,
+> {
+	/**
+	 * Kind of {@link TreeNodeSchema} this JSON Schema entry represents.
+	 *
+	 * @remarks There is not a 1:1 mapping between {@link TreeNodeSchema} types and JSON Schema types.
+	 * This is used to disambiguate the type of {@link TreeNodeSchema} this JSON Schema maps to.
+	 *
+	 * Note: This property name is not a part of the JSON Schema spec. This is a Fluid-specific extension.
+	 */
+	readonly _kind: TNodeKind;
+
+	/**
+	 * {@inheritDoc JsonSchemaType}
+	 */
+	readonly type: TJsonSchemaType;
+}
+
+/**
+ * JSON Schema for an object node.
+ * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-instance-data-model}.
+ * @alpha
+ */
+export interface ObjectNodeJsonSchema extends NodeJsonSchemaBase<"object", "object"> {
+	/**
+	 * Object fields.
+	 * @remarks Required fields should have a corresponding entry in {@link ObjectNodeJsonSchema.required}.
+	 * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-properties}.
+	 */
+	readonly properties: Record<string, FieldJsonSchema>;
+
+	/**
+	 * List of keys for required fields.
+	 *
+	 * @remarks
+	 * Optional fields should not be included in this list.
+	 * Each key specified must have an entry in {@link ObjectNodeJsonSchema.properties}.
+	 */
+	readonly required?: string[];
+
+	/**
+	 * Whether or not additional properties (properties not specified by the schema) are allowed in objects of this type.
+	 * @see {@link https://json-schema.org/draft/2020-12/json-schema-core#name-additionalproperties}.
+	 */
+	readonly additionalProperties?: boolean;
```

Comment:
How does this relate to `_kind`? Is `_kind` not an additional property? If it is, then doesn't "additionalProperties" always need to be true? Is it reasonable to expect that some other system that we might dump our JSON schema object to would fail if it sees `_kind` but not `additionalProperties`? There is a note below about external tools and opting out of "strict" validation - but could "strict" validation still succeed against `_kind` if "additionalProperties" were enabled?

Reply:
`_kind` is a property on the JSON Schema object, _not_ on the data that would be inserted into a tree. `additionalProperties` refers to whether or not additional properties are allowed on the _data_ corresponding with this schema.